public/pages/ChangePolicy/containers/ChangePolicy/ChangePolicy.tsx

```diff
@@ -53,6 +55,22 @@ export default class ChangePolicy extends Component<ChangePolicyProps, ChangePol
     this.context.chrome.setBreadcrumbs([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.MANAGED_INDICES, BREADCRUMBS.CHANGE_POLICY]);
   }
 
+  componentDidUpdate(prevProps: ChangePolicyProps, prevState: Readonly<ChangePolicyState>) {
+    if (prevProps.dataSourceId !== this.props.dataSourceId) {
+      // reset the state, if dataSourceId changes, i.e., clear state
+      this.setState({
+        selectedPolicies: [],
+        selectedManagedIndices: [],
+        selectedStateFilters: [],
+        stateRadioIdSelected: Radio.Current,
+        stateSelected: "",
+        managedIndicesError: "",
+        selectedPoliciesError: "",
+        hasSubmitted: false,
+      });
```

Comment:
Nit: consider to extract as variable like default state?

Reply:
Done